include/revng/Pipeline/AnyDiff.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <any>
+
+#include "revng/Pipeline/InvalidationEvent.h"
+#include "revng/Pipeline/TupleTreeInvalidationEvent.h"
+#include "revng/Support/YAMLTraits.h"
+#include "revng/TupleTree/TupleTreeDiff.h"
+
+namespace pipeline {
+class AnyDiffBase {
+public:
+  virtual void serialize(llvm::raw_ostream &OS) const = 0;
+  virtual std::unique_ptr<InvalidationEventBase>
+  getInvalidationEvent() const = 0;
```

Comment:
Does a diff *needs* to be associated with an existing invalidation event? Or is this *creating* a new invalidation event?
In the latter case, `createInvalidationEvent` might be a more proper name.

Reply:
well, i mean, it returns a unique pointer, it must be a new one every time

Reply:
Yeah, you're right, but I think `create` gives a more specific hint to the fact that we're not getting something that already existed but we're creating it from scratch.